src/gpu/gpu_deconvolution_list.cpp

```diff
@@ -45,18 +48,18 @@ using namespace dnnl::impl::prop_kind;
 const std::map<pk_impl_key_t, std::vector<impl_list_item_t>>
         impl_list_map REG_DECONV_P({
     {{forward}, {
-        GPU_INSTANCE_INTEL(intel::ocl::convolution_deconvolution_fwd_t)
         GPU_INSTANCE_NVIDIA(nvidia::cudnn_deconvolution_fwd_t)
         GPU_INSTANCE_AMD(amd::miopen_deconvolution_fwd_t)
+        GPU_INSTANCE_GENERIC(generic::convolution_deconvolution_fwd_t)
         nullptr,
     }},
     {{backward}, REG_BWD_PK({
-        GPU_INSTANCE_INTEL(intel::ocl::convolution_deconvolution_bwd_data_t)
         GPU_INSTANCE_INTEL(intel::ocl::convolution_deconvolution_bwd_weights_t)
         GPU_INSTANCE_NVIDIA(nvidia::cudnn_deconvolution_bwd_data_t)
         GPU_INSTANCE_NVIDIA(nvidia::cudnn_deconvolution_bwd_weights_t)
         GPU_INSTANCE_AMD(amd::miopen_deconvolution_bwd_data_t)
         GPU_INSTANCE_AMD(amd::miopen_deconvolution_bwd_weights_t)
+        GPU_INSTANCE_GENERIC(generic::convolution_deconvolution_bwd_data_t)
```

Comment:
Does this have a performance impact for Intel? If there's a case where both would match, it would now match `data` before `weights`.

Reply:
With these specific implementations, there should be no performance impact. They're mutually exclusive, looking for `dnnl_backward_weights` and `dnnl_backward_data` propagation kinds, respectively.